### PR TITLE
project,project_issue,portal_project: cleaning code.

### PR DESCRIPTION
**Pull request by atp-odoo**
[IMP]: Clean some code from project,project_issue (preapi)
Task : https://www.odoo.com/web?#id=8847&view_type=form&model=project.task&action=327
pad : https://pad.odoo.com/p/r.kZxs1A0VXfYsHCv2

### CODE REVIEW

File: addons/portal_project/security/ir.model.access.csv [hunk 1]
```diff
@@ -3,11 +3,11 @@ access_project,project,project.model_project_project,base.group_portal,1,0,0,0
 access_task,task,project.model_project_task,base.group_portal,1,0,0,0
 access_task_type,task_type,project.model_project_task_type,base.group_portal,1,0,0,0
 access_task_work,task_work,project.model_project_task_work,base.group_portal,1,0,0,0
-access_project_category,project_category,project.model_project_category,base.group_portal,1,0,0,0
+access_project_tags,project_tags,project.model_project_tags,base.group_portal,1,0,0,0
```
Comment (tde-banana-odoo) [inline addons/portal_project/security/ir.model.access.csv]: ID: access_project_tags_portal
Name: project.tags.portal

File: addons/portal_project/security/ir.model.access.csv [hunk 2]
```diff
@@ -3,11 +3,11 @@ access_project,project,project.model_project_project,base.group_portal,1,0,0,0
...
 access_account_analytic_account,account_analytic_account,analytic.model_account_analytic_account,base.group_portal,1,0,0,0
 access_project_public,project,project.model_project_project,base.group_public,1,0,0,0
 access_task_public,task,project.model_project_task,base.group_public,1,0,0,0
 access_task_type_public,task_type,project.model_project_task_type,base.group_public,1,0,0,0
 access_task_work_public,task_work,project.model_project_task_work,base.group_public,1,0,0,0
-access_project_category_public,project_category,project.model_project_category,base.group_public,1,0,0,0
+access_project_tags_public,project_tags,project.model_project_tags,base.group_public,1,0,0,0
```
Comment (tde-banana-odoo) [inline addons/portal_project/security/ir.model.access.csv]: Regroup by model + see above remark

File: addons/project/project.py [hunk 3]
```diff
@@ -62,15 +62,9 @@ class project_task_type(osv.osv):
...
-    def _get_default_project_ids(self, cr, uid, ctx={}):
-        project_id = self.pool['project.task']._get_default_project_id(cr, uid, context=ctx)
-        if project_id:
-            return [project_id]
-        return None
-
     _defaults = {
         'sequence': 1,
-        'project_ids': _get_default_project_ids,
+        'project_ids': lambda self, cr, uid, ctx: [ctx['default_project_id']] if ctx.get('default_project_id') else None
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Context can be None. Keep the method and put this code in the default + context check.

File: addons/project/project.py [hunk 4]
```diff
@@ -645,37 +604,15 @@ class task(osv.osv):
     }
 
     def _get_default_partner(self, cr, uid, context=None):
-        project_id = self._get_default_project_id(cr, uid, context)
-        if project_id:
-            project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
+        if context.get('default_project_id'):
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Check that context exists.

File: addons/project/project.py [hunk 5]
```diff
@@ -645,37 +604,15 @@ class task(osv.osv):
...
-            context = {}
-        if type(context.get('default_project_id')) in (int, long):
-            return context['default_project_id']
-        if isinstance(context.get('default_project_id'), basestring):
-            project_name = context['default_project_id']
-            project_ids = self.pool.get('project.project').name_search(cr, uid, name=project_name, context=context)
-            if len(project_ids) == 1:
-                return project_ids[0][0]
-        return None
+        return self.stage_find(cr, uid, [], context.get('default_project_id'), [('fold', '=', False)], context=context)
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Check that context exists.

File: addons/project/project.py [hunk 6]
```diff
@@ -684,9 +621,8 @@ def _read_group_stage_ids(self, cr, uid, ids, domain, read_group_order=None, acc
         if read_group_order == 'stage_id desc':
             order = '%s desc' % order
         search_domain = []
-        project_id = self._resolve_project_id_from_context(cr, uid, context=context)
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Call the default method and use project_id hereafter.

File: addons/project/project.py [hunk 7]
```diff
@@ -701,10 +637,9 @@ def _read_group_stage_ids(self, cr, uid, ids, domain, read_group_order=None, acc
 
     def _read_group_user_id(self, cr, uid, ids, domain, read_group_order=None, access_rights_uid=None, context=None):
         res_users = self.pool.get('res.users')
-        project_id = self._resolve_project_id_from_context(cr, uid, context=context)
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Same remark as above.

File: addons/project/project.py [hunk 8]
```diff
@@ -701,10 +637,9 @@ def _read_group_stage_ids(self, cr, uid, ids, domain, read_group_order=None, acc
...
         access_rights_uid = access_rights_uid or uid
-        if project_id:
-            ids += self.pool.get('project.project').read(cr, access_rights_uid, project_id, ['members'], context=context)['members']
+        if context.get('default_project_id'):
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Check that context exists.

File: addons/project/project.py [hunk 9]
```diff
@@ -684,9 +621,8 @@ def _read_group_stage_ids(self, cr, uid, ids, domain, read_group_order=None, acc
         if read_group_order == 'stage_id desc':
             order = '%s desc' % order
         search_domain = []
-        project_id = self._resolve_project_id_from_context(cr, uid, context=context)
-        if project_id:
-            search_domain += ['|', ('project_ids', '=', project_id), ('id', 'in', ids)]
+        if context.get('default_project_id'):
+            search_domain += ['|', ('project_ids', '=', context['default_project_id']), ('id', 'in', ids)]
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Check that context exists.

File: addons/project/project.py [hunk 10]
```diff
@@ -803,7 +733,7 @@ def _get_task(self, cr, uid, ids, context=None):
         'sequence': fields.integer('Sequence', select=True, help="Gives the sequence order when displaying a list of tasks."),
         'stage_id': fields.many2one('project.task.type', 'Stage', track_visibility='onchange', select=True,
                         domain="[('project_ids', '=', project_id)]", copy=False),
-        'categ_ids': fields.many2many('project.category', string='Tags'),
+        'tag_ids': fields.many2many('project.tags', string='Tags'),
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Use oldname.

File: addons/project/project.py [hunk 11]
```diff
@@ -855,7 +785,7 @@ def _get_task(self, cr, uid, ids, context=None):
     }
     _defaults = {
         'stage_id': _get_default_stage_id,
-        'project_id': _get_default_project_id,
+        'project_id': lambda self, cr, uid, ctx=None: ctx.get('default_project_id'),
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: Check that context exists.

File: addons/project/project_demo.xml [hunk 12]
```diff
@@ -8,16 +8,16 @@
         </record>
 
         <!-- Categories -->
-        <record id="project_category_01" model="project.category">
+        <record id="project_tags_01" model="project.tags">
```
Comment (tde-banana-odoo) [inline addons/project/project_demo.xml]: Begin at _0 .

File: addons/project/security/ir.model.access.csv [hunk 13]
```diff
@@ -22,6 +22,6 @@ access_project_task_history_cumulative,project.task.history project,project.mode
 access_resource_calendar,project.resource_calendar user,resource.model_resource_calendar,project.group_project_user,1,0,0,0
 access_resource_calendar_attendance,project.resource_calendar_attendance user,resource.model_resource_calendar_attendance,project.group_project_user,1,0,0,0
 access_resource_calendar_leaves_user,resource.calendar.leaves user,resource.model_resource_calendar_leaves,project.group_project_user,1,1,1,1
-access_project_category,project.project_category,model_project_category,,1,0,0,0
-access_project_category_manager,project.project_category,model_project_category,project.group_project_manager,1,1,1,1
+access_project_tags,project.project_tags,model_project_tags,,1,0,0,0
```
Comment (tde-banana-odoo) [inline addons/project/security/ir.model.access.csv]: access_project_tags_all,project.task.all, ... (see above remark + propagate)

File: addons/portal_project/security/ir.model.access.csv [hunk 14]
```diff
@@ -3,11 +3,11 @@ access_project,project,project.model_project_project,base.group_portal,1,0,0,0
 access_task,task,project.model_project_task,base.group_portal,1,0,0,0
```
Comment (tde-banana-odoo) [inline addons/portal_project/security/ir.model.access.csv]: All rules related to portal / public can be moved directly into project, as the groups are now defined in base.

File: addons/project_issue/project_issue.py [hunk 15]
```diff
@@ -68,37 +68,15 @@ class project_issue(osv.Model):
     }
 
     def _get_default_partner(self, cr, uid, context=None):
-        project_id = self._get_default_project_id(cr, uid, context)
-        if project_id:
-            project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
+        if context.get('default_project_id'):
```
Comment (tde-banana-odoo) [inline addons/project_issue/project_issue.py]: Check that context exists. Okay, you got the message, do it everywhere it is necessary :) .

File: addons/project_issue/project_issue_demo.xml [hunk 16]
```diff
@@ -25,13 +25,13 @@
     <data noupdate="1">
 
         <!-- Categories -->
-        <record id="project_issue_category_01" model="project.category">
+        <record id="project_issue_tags_01" model="project.tags">
```
Comment (tde-banana-odoo) [inline addons/project_issue/project_issue_demo.xml]: As above, begin at _0.

File: addons/portal_project/security/ir.model.access.csv (hunk 1 above)
Comment (ptr-odoo) [inline addons/portal_project/security/ir.model.access.csv]: Done.

File: addons/portal_project/security/ir.model.access.csv (hunk 2 above)
Comment (ptr-odoo) [inline addons/portal_project/security/ir.model.access.csv]: Done.

File: addons/project/project.py (hunk 3 above)
Comment (ptr-odoo) [inline addons/project/project.py]: Done.

File: addons/project/project.py (hunk 4 above)
Comment (ptr-odoo) [inline addons/project/project.py]: Done.

File: addons/project/project.py (hunk 5 above)
Comment (ptr-odoo) [inline addons/project/project.py]: Done.

File: addons/project/project.py (hunk 8 above)
Comment (ptr-odoo) [inline addons/project/project.py]: Done.

File: addons/project/project.py (hunk 9 above)
Comment (ptr-odoo) [inline addons/project/project.py]: Done.

File: addons/project/project.py (hunk 11 above)
Comment (ptr-odoo) [inline addons/project/project.py]: Done.

File: addons/project/project_demo.xml (hunk 12 above)
Comment (ptr-odoo) [inline addons/project/project_demo.xml]: Done.

File: addons/project/security/ir.model.access.csv (hunk 13 above)
Comment (ptr-odoo) [inline addons/project/security/ir.model.access.csv]: Done.

File: addons/portal_project/security/ir.model.access.csv (hunk 14 above)
Comment (ptr-odoo) [inline addons/portal_project/security/ir.model.access.csv]: Done.

**Comment by rim-odoo** [comment]
@ptr-odoo Can you rebase and fix the conflicts? Thanks

File: addons/project/project.py [hunk 17]
```diff
@@ -60,10 +60,10 @@ class project_task_type(osv.osv):
                                'there are no records in that stage to display.'),
     }
 
-    def _get_default_project_ids(self, cr, uid, ctx={}):
-        project_id = self.pool['project.task']._get_default_project_id(cr, uid, context=ctx)
-        if project_id:
-            return [project_id]
+    def _get_default_project_ids(self, cr, uid, ctx=None):
+        ctx = dict(ctx or {})
```
Comment (rim-odoo) [inline addons/project/project.py]:
```python 
if cxt is None:
    ctx = {}
```

File: addons/project/project.py [hunk 18]
```diff
@@ -615,48 +580,28 @@ class task(osv.osv):
     _mail_post_access = 'read'
 
     def _get_default_partner(self, cr, uid, context=None):
-        project_id = self._get_default_project_id(cr, uid, context)
-        if project_id:
-            project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
+        context = dict(context or {})
```
Comment (rim-odoo) [inline addons/project/project.py]:
```python 
if cxt is None:
    ctx = {}
```

File: addons/project/project.py [hunk 19]
```diff
@@ -615,48 +580,28 @@ class task(osv.osv):
...
-            context = {}
-        if type(context.get('default_project_id')) in (int, long):
-            return context['default_project_id']
-        if isinstance(context.get('default_project_id'), basestring):
-            project_name = context['default_project_id']
-            project_ids = self.pool.get('project.project').name_search(cr, uid, name=project_name, context=context)
-            if len(project_ids) == 1:
-                return project_ids[0][0]
-        return None
+        context = dict(context or {})
```
Comment (rim-odoo) [inline addons/project/project.py]:
```python 
if cxt is None:
    ctx = {}
```

File: addons/project/project.py [hunk 20]
```diff
@@ -837,6 +777,7 @@ def _get_task(self, cr, uid, ids, context=None):
         'partner_id': lambda self, cr, uid, ctx=None: self._get_default_partner(cr, uid, context=ctx),
         'date_start': fields.datetime.now,
     }
+
```
Comment (rim-odoo) [inline addons/project/project.py]: unuseful diff

File: addons/project/project.py [hunk 21]
```diff
@@ -1408,9 +1330,9 @@ def init(self, cr):
         )
         """)
 
-class project_category(osv.osv):
+class project_tags(osv.osv):
```
Comment (rim-odoo) [inline addons/project/project.py]: osv.Model

File: addons/project/tests/test_project_flow.py [hunk 22]
```diff
@@ -129,4 +116,4 @@ def test_task_process(self):
                 'planned_hours_me': 2.0,
             }).with_context({'active_id': task.id}).delegate()
         # Check delegation details
-        self.assertEqual(task.planned_hours, 2, 'project_task_delegate: planned hours is not correct after delegation')
+        self.assertEqual(task.planned_hours, 2, 'project_task_delegate: planned hours is not correct after delegation')
```
Comment (rim-odoo) [inline addons/project/tests/test_project_flow.py]: add line return

File: addons/project/project.py (hunk 17 above)
Comment (ptr-odoo) [inline addons/project/project.py]: @rim-odoo : `ctx = dict(ctx or {})` and  `if cxt is None: ctx = {}` they are retrun same value.

Comment (rim-odoo) [inline addons/project/project.py]: yes, but if ctx == {}, then you prevent recreating a new {}

**Comment by rim-odoo** [comment]
@atp-odoo Sorry for the delay, could you fix the conflicts? Thanks a lot

**Comment by atp-odoo** [comment]
Hello @rim-odoo, no problem I resolved conflicts. 

Thanks

**Comment by rim-odoo** [comment]
@atp-odoo Thanks Atul

File: addons/project/project.py [hunk 23]
```diff
@@ -60,10 +60,11 @@ class project_task_type(osv.osv):
...
-    def _get_default_project_ids(self, cr, uid, ctx={}):
-        project_id = self.pool['project.task']._get_default_project_id(cr, uid, context=ctx)
-        if project_id:
-            return [project_id]
+    def _get_default_project_ids(self, cr, uid, ctx=None):
+        if ctx is None:
+            ctx = {}
+        if ctx.get('default_project_id'):
+            return [ctx['default_project_id']]
         return None
```
Comment (rim-odoo) [inline addons/project/project.py]: default_project_id = ctx.get('default_project_id')
return [default_project_id] if default_project_id else None

File: addons/project/project.py [hunk 24]
```diff
@@ -615,48 +581,31 @@ class task(osv.osv):
     _mail_post_access = 'read'
 
     def _get_default_partner(self, cr, uid, context=None):
-        project_id = self._get_default_project_id(cr, uid, context)
-        if project_id:
-            project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
+        if context is None:
+            context = {}
+        if context.get('default_project_id'):
```
Comment (rim-odoo) [inline addons/project/project.py]: If you simply want to check that key `default_project_id` is in dictionary `context`, you can use `in` operator; `get` method reads the value associated to the key.
```python
if 'default_project_id' in context:
```

File: addons/project/project.py [hunk 25]
```diff
@@ -615,48 +581,31 @@ class task(osv.osv):
...
         stage_obj = self.pool.get('project.task.type')
         order = stage_obj._order
         access_rights_uid = access_rights_uid or uid
         if read_group_order == 'stage_id desc':
             order = '%s desc' % order
         search_domain = []
-        project_id = self._resolve_project_id_from_context(cr, uid, context=context)
-        if project_id:
-            search_domain += ['|', ('project_ids', '=', project_id), ('id', 'in', ids)]
+        if context.get('default_project_id'):
```
Comment (rim-odoo) [inline addons/project/project.py]: Same remark here

File: addons/project/project.py (hunk 8 above)
Comment (rim-odoo) [inline addons/project/project.py]: Same remark here

File: addons/project_issue/project_issue.py [hunk 26]
```diff
@@ -53,39 +53,23 @@ class project_issue(osv.Model):
     _mail_post_access = 'read'
 
     def _get_default_partner(self, cr, uid, context=None):
-        project_id = self._get_default_project_id(cr, uid, context)
-        if project_id:
-            project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
+        if context is None:
+            context = {}
+        if context.get('default_project_id'):
```
Comment (rim-odoo) [inline addons/project_issue/project_issue.py]: same remark here

File: addons/project_issue/project_issue.py [hunk 27]
```diff
@@ -97,9 +81,8 @@ def _read_group_stage_ids(self, cr, uid, ids, domain, read_group_order=None, acc
         # - OR ('case_default', '=', True), ('fold', '=', False): add default columns that are not folded
         # - OR ('project_ids', 'in', project_id), ('fold', '=', False) if project_id: add project columns that are not folded
         search_domain = []
-        project_id = self._resolve_project_id_from_context(cr, uid, context=context)
-        if project_id:
-            search_domain += ['|', ('project_ids', '=', project_id), ('id', 'in', ids)]
+        if context.get('default_project_id'):
```
Comment (rim-odoo) [inline addons/project_issue/project_issue.py]: same remark here

**Comment by rim-odoo** [comment]
@atp-odoo @ptr-odoo When I launch this branch and I install project, project_issue and project_portal, I get this warning:
```python
2015-04-22 13:48:05,658 14572 WARNING master openerp.modules.loading: The model account.analytic.journal has no access rules, consider adding one. E.g. access_account_analytic_journal,access_account_analytic_journal,model_account_analytic_journal,,1,0,0,0
```

**Comment by atp-odoo** [comment]
Hello @rim-odoo ,
This warning is generated when we install only project or analytic module.
This warning is also generated in master branch.
Problem is when we install project module, project is depended on analytic module and it's going to install analytic module. object account.analytic.journal is defined on this analytic module but there is not access rights , ir.model.access.csv file is emply. so if we defined rule in this module for account.analytic.journal object then this warning is not raised. now we will fix  issue in this branch.
Thanks

File: addons/analytic/security/ir.model.access.csv [hunk 28]
```diff
@@ -1 +1,2 @@
-id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
\ No newline at end of file
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_account_analytic_journal,account.analytic.journal,model_account_analytic_journal,,1,1,1,1
```
Comment (rim-odoo) [inline addons/analytic/security/ir.model.access.csv]: seriously?

Comment (atp-odoo) [inline addons/analytic/security/ir.model.access.csv]: ok. yes corrected. To remove warning we can use readonly access.

**Pull request closed without merging**